.github/workflows/default.yml

```diff
@@ -0,0 +1,289 @@
+        image: mysql:8.0
+        env:
+          MYSQL_USER: links
+          MYSQL_USER_PASSWORD: links
+          MYSQL_ROOT_PASSWORD: links
+          MYSQL_DATABASE: links
+          MYSQL_ROOT_HOST: '%'
+        ports:
+          - 3307:3306
+        options: >-
+          --health-cmd="mysqladmin ping"
+          --health-interval=10s
+          --health-timeout=5s
+          --health-retries=3
+
+    steps:
+      - name: Configure MySQL
+        run: |
+          mysql --version
+          sudo systemctl enable mysql
+          sudo systemctl start mysql
+          mysql -h 127.0.0.1 -P 3307 -uroot -plinks -e "CREATE USER links IDENTIFIED BY 'links';"
+          mysql -h 127.0.0.1 -P 3307 -uroot -plinks -e "GRANT ALL ON  links.* TO links;"
+          mysql -h 127.0.0.1 -P 3307 -uroot -plinks -e "FLUSH PRIVILEGES;"
+        shell: bash
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install system dependencies
+        run: sudo apt-get install -y libev-dev
+        shell: bash
+
+      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+
+      - run: opam install -y conf-libev
+      - run: opam install -y ounit2
+
+      - name: Install Links dependencies
+        run: >-
+          opam install
+          ./links.opam
+          ./links-postgresql.opam
+          ./links-mysql8.opam
+          ./links-sqlite3.opam
+          --deps-only
+
+      - name: Build Links from source
+        run: |
+          eval $(opam env)
+          make all-ci
+        shell: bash
+
+      - name: Run database testsuite
+        run: |
+          eval $(opam env)
+          ./run-database-tests tests/database -d all
+          ./run-database-tests tests/shredding -d all
+          ./run-database-tests tests/relational-lenses -d all
+          ./run-tests unit
+        env:
+          # TODO(dhil): I do not how to index into the environments of
+          # either service. Until we figure out how to do it, we will
+          # just duplicate information here.
+          LINKS_POSTGRES_HOST: localhost
+          LINKS_POSTGRES_USER: links
+          LINKS_POSTGRES_PASSWORD: links
+          LINKS_POSTGRES_PORT: ${{ job.services.postgres.ports[5432] }}
+          LINKS_MYSQL8_HOST: 127.0.0.1
+          LINKS_MYSQL8_USER: links
+          LINKS_MYSQL8_PASSWORD: links
+          LINKS_MYSQL8_PORT: ${{ job.services.mysql.ports[3306] }}
+        shell: bash
+
+  mysql5-7:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-18.04
+        ocaml-compiler:
+          - 4.08.0
+
+    runs-on: ${{ matrix.os }}
+
+    services:
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_USER: links
+          MYSQL_USER_PASSWORD: links
+          MYSQL_ROOT_PASSWORD: links
+          MYSQL_DATABASE: links
+          MYSQL_ROOT_HOST: '%'
+        ports:
+          - 3308:3306
```

Comment:
I would suggest mapping to the same standard MySQL port 3306.

Reply:
I did that initially, but it doesn't work as the host image already occupies that port.